### PR TITLE
Doc Tweaks: Grammar, clarification, detail

### PR DESCRIPTION
**Pull request by actionbuddha**
The following PR makes a few minor modifications to the docs, adding some ports and detail, moving things around to increase consistency, and a few grammatical modifications. I particularly wanted to be a little more explicit about some of our prereqs - naming kernel modules required, etc..

### CODE REVIEW

File: _docs/prerequisites/devicepresentation.md [hunk 1]
```diff
@@ -7,10 +7,22 @@ module: prerequisites/devicepresentation
...
 # Device presentation
 
-Certain kernel modules are required to use StorageOS, in particular Linux-IO
-(LIO), an open-source implementation of the SCSI target. LIO support depends
-on the operating system.
+StorageOS requires certain kernel modules to function, in particular [Linux-IO
+](http://linux-iscsi.org/wiki/Main_Page), an open-source implementation of the
+SCSI target.
 
+We require the following modules to be loaded:
```
[the remark this answers is not in the document]
Comment (actionbuddha) [reply _docs/prerequisites/devicepresentation.md]: Ack, will modify.

File: _docs/prerequisites/firewalls.md [hunk 2]
```diff
@@ -23,34 +24,33 @@ StorageOS requires a few ports to be open:
...
-iptables -A INPUT -p udp --dport 5711 -j ACCEPT
+# Inbound traffic
+iptables -I INPUT -i lo -m comment --comment 'Permit loopback traffic' -j ACCEPT
+iptables -I INPUT -m state --state ESTABLISHED,RELATED -m comment --comment 'Permit established traffic' -j ACCEPT
+iptables -A INPUT -p tcp --dport 5701:5711 -m comment --comment 'StorageOS' -j ACCEPT
+iptables -A INPUT -p udp --dport 5711 -m comment --comment 'StorageOS' -j ACCEPT
+
+# Outbound traffic
+iptables -I OUTPUT -o lo -m comment --comment 'Permit loopback traffic' -j ACCEPT
+iptables -I OUTPUT -d 0.0.0.0/0 -m comment --comment 'Permit outbound traffic' -j ACCEPT
```
[the remark this answers is not in the document]
Comment (Arau) [reply _docs/prerequisites/firewalls.md]: Done, approved and merged

File: _docs/prerequisites/firewalls.md [hunk 3]
```diff
@@ -23,34 +24,51 @@ StorageOS requires a few ports to be open:
 | 5710        | tcp       | NATS service           |
 | 5711        | tcp & udp | Gossip service         |
 
-StorageOS also uses random ports to dial-out to these ports on other StorageOS nodes. For this reason, outgoing traffic should be enabled.
+StorageOS also uses [ephemeral](https://en.wikipedia.org/wiki/Ephemeral_port) ports to dial-out to these ports on other StorageOS nodes. For this reason, outgoing traffic should be enabled.
```
Comment (croomes) [inline _docs/prerequisites/firewalls.md]: Not related to this PR, but 5701 and 5702 are internal-only and no inter-node connectivity is required.

Comment (croomes) [reply _docs/prerequisites/firewalls.md]: Same with 5704

Comment (croomes) [reply _docs/prerequisites/firewalls.md]: Might just want to mark them as "reserved"

File: _docs/prerequisites/mountpropagation.md [hunk 4]
```diff
@@ -8,17 +8,39 @@ module: prerequisites/mountpropagation
...
-not be enabled by default.
+containers (see linux kernel documentation
+[here](http://man7.org/linux/man-pages/man2/mount.2.html)).
 
-On the other hand, orchestrators such as Kubernetes or OpenShift have their own
-implementations. K8S 1.10 and OpenShift 3.10 have mount propagation enabled by
+Certain versions of docker ship with a systemd manifest with
+[MountFlags](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#) set
+to 'slave', thus preventing StorageOS from working. This can be removed or set
+to 'shared' with a simple systemd drop in:
```
Comment (croomes) [inline _docs/prerequisites/mountpropagation.md]: I'd remove "simple"

File: _docs/prerequisites/mountpropagation.md [hunk 5]
```diff
@@ -8,17 +8,39 @@ module: prerequisites/mountpropagation
...
+# systemctl restart docker.service
+```
+
+To confirm behaviour, the following command should run without error
+
+```
+docker run -it --rm -v /mnt:/mnt:shared busybox sh -c /bin/date
+```
+
+Orchestrators such as kubernetes or openshift have their own ways of exposing
```
Comment (croomes) [inline _docs/prerequisites/mountpropagation.md]: Capitalise "Kubernetes or Openshift"

Comment (croomes) [reply _docs/prerequisites/mountpropagation.md]: Actually, Openshift should be "OpenShift"

File: _docs/prerequisites/firewalls.md (hunk 3 above)
Comment (actionbuddha) [reply _docs/prerequisites/firewalls.md]: I'd argue explicit is better - even intra-host comms will show via netstat/tcpdump/etc. and it's the sort of detail appreciated by some.

File: _docs/prerequisites/mountpropagation.md (hunk 5 above)
Comment (actionbuddha) [reply _docs/prerequisites/mountpropagation.md]: Pushed a commit which makes these consistent across the site.

**Comment by actionbuddha** [comment]
I think we're about ready here - good to merge?

**Pull request merged**